File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        //Approach 1
        int[] arr= new int[nums.length];
        for(int i =0;i<nums.length;i++){
            arr[i]=nums[i];
        }
        Arrays.sort(arr);
        int start =0; int end = nums.length-1;
        for(;start<nums.length;start++){
            if(nums[start] != arr[start])break;
        }
        if(start>=nums.length -1)return 0;
        for(;end>=0;end--){
            if(nums[end] !=arr[end])break;
        }
        return end-start +1;  
    }
}

File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
      int end =-1;
      int ele = nums[0];
      for(int i =1;i<nums.size();i++){
          if(ele>nums[i]){
              end = i;
          }else{
              ele = nums[i];
          }
      }
        int st = 0;
        ele = nums[nums.size() -1];
        for(int i =nums.size() -2;i>=0;i--){
           if(nums[i] >ele){
               st = i;
           }else{
               ele = nums[i]; 
           }
    
        }
    return end - st  +1;
    }
};